src/js/lib/api/mock.js

```diff
@@ -57,5 +56,28 @@ export function updateUserInfo(id, newUserInfo, jwt, callback) {
     return setTimeout(() => {
         userJSONObj = Object.assign({}, userJSONObj, newUserInfo);
         return callback(null, userJSONObj);
-    });
+    }, LATENCY);
+}
+
+const mockGeneData = {
+    myGene1: {
+        name: "myGene1",
+        id: 4
+    }
+};
+
+/**
+ * Get a single gene by it's name
+ * @param  {String}   name     - the name of the gene
+ * @param  {Function} callback - the callback
+ */
+export function getGeneByFullName(name, callback) {
+    return setTimeout(() => {
+        if(name in mockGeneData) {
+            return callback(null, mockGeneData[name]);
+        }
+
+        return callback({error: "TooManyGenes"});
```

Comment:
This part of the code is basically on hold until we get the gene format stuff sorted out. Originally this is when we discussed an array being returned as the results. If the array contained more than one element it would return a TooManyGenes error. But again this is all on hold for now so whats here will likely change significantly.